### PR TITLE
[testing ci] add tests for unicode tarballs, fix utf-8 on macOS CI

### PR DESCRIPTION
**Pull request by strega-nil**
see the errors in #19963

### CODE REVIEW

**Comment by BillyONeal** [comment]
I think this belongs as an e2e test rather than here?

**Comment by strega-nil-ms** [comment]
@BillyONeal it's testing `vcpkg_extract_source_archive`, so having it as a port test is fine, no?

**Comment by BillyONeal** [comment]
> @BillyONeal it's testing `vcpkg_extract_source_archive`, so having it as a port test is fine, no?

Kinda related to https://github.com/microsoft/vcpkg/pull/19034 where I made the same comment. As far as I know this is testing scripts/cmake meaning it logically belongs in the tool repo.

**Comment by strega-nil-ms** [comment]
@BillyONeal At that point, we should probably pull out the test_ports directory, but until then the scripts still live in `vcpkg`, right?

**Comment by dg0yt** [comment]
>  At that point, we should probably pull out the test_ports directory, but until then the scripts still live in `vcpkg`, right?

Note the table I posted in https://github.com/microsoft/vcpkg/pull/19034#issuecomment-913354954. IMO key question is what triggers the test. If scripts are moved out of `vcpkg`, #20060 can be moved out, too. But some tests in `test_ports` are really meant to be triggered by changes to ports (i.e. to `vcpkg` repo), in order to ensure usability of the ports or selected non-default feature combinations.

File: scripts/azure-pipelines/osx/azure-pipelines.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ jobs:
     clean: resources
   timeoutInMinutes: 1440 # 1 day
   variables:
+  - name: LC_ALL
```
Comment (BillyONeal) [inline scripts/azure-pipelines/osx/azure-pipelines.yml:13]: It seems like maybe vcpkg itself should be setting this rather than hiding it in CI? It would be in keeping with other environment variable handling we do....

Comment (strega-nil-ms) [reply scripts/azure-pipelines/osx/azure-pipelines.yml:13]: I don't know how to set this variable in general; I don't even know if it's possible. Perhaps we could do some locale parsing; however, for CI, it makes sense that we set it to en_US.UTF-8.

**Comment by BillyONeal** [comment]
I think the correct fix here is to have vcpkg set the locale by testing a few common ones like:

* `C.UTF-8`
* `POSIX.UTF-8`
* `en_US.UTF-8`

and choosing the first matching one.

Notably, this change plus #19963 will break existing macos customers, most of whom aren't passing LC_ALL like this.

**Comment by JackBoosY** [comment]
So, should we wait for the next vcpkg release?
